source/exe/posix/platform_main.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#include "common/common/macros.h"
+#include "common/common/thread_impl.h"
+
+namespace Envoy {
+
+class PlatformMain {
```

Comment:
I'm still uncomfortable with this naming. `PlatformMain` to me should be a method, it's essentially an entry point, not a noun. Can we just make this `PlatformImpl`?

Reply:
LGTM sans this bikeshed :)